src/screens/SendSynthetic/SendSyntheticAsset.js

```diff
@@ -0,0 +1,240 @@
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { FlatList } from 'react-native';
+import { NavigationScreenProp } from 'react-navigation';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import styled from 'styled-components/native';
+import get from 'lodash.get';
+import isEmpty from 'lodash.isempty';
+import { SDK_PROVIDER } from 'react-native-dotenv';
+
+// components
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import ListItemWithImage from 'components/ListItem/ListItemWithImage';
+import Separator from 'components/Separator';
+import { Container, Wrapper } from 'components/Layout';
+import EmptyStateParagraph from 'components/EmptyState/EmptyStateParagraph';
+import { getAssetData, getAssetsAsList } from 'utils/assets';
+import Spinner from 'components/Spinner';
+
+// actions
+import { initSyntheticsServiceAction } from 'actions/syntheticsActions';
+
+// utils, services
+import { spacing, UIColors } from 'utils/variables';
+import syntheticsService from 'services/synthetics';
+import { makePromiseCancelable } from 'utils/common';
+
+// constants
+import { SEND_TOKEN_CONTACTS } from 'constants/navigationConstants';
+import { PLR } from 'constants/assetsConstants';
+
+// selectors
+import { accountAssetsSelector } from 'selectors/assets';
+import { availableStakeSelector } from 'selectors/paymentNetwork';
+
+// models, types
+import type { Asset, Assets } from 'models/Asset';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+
+// configs
+import assetsConfig from 'configs/assetsConfig';
+
+type Props = {
+  accountAssets: Assets,
+  supportedAssets: Asset[],
+  initSyntheticsService: () => void,
+  navigation: NavigationScreenProp<*>,
+  availableStake: number,
+};
+
+type State = {
+  loadingAssets: boolean,
+  availableAssets: [],
+};
+
+const InnerWrapper = styled(Wrapper)`
+  flex: 1;
+  margin-top: ${spacing.large}px;
+`;
+
+const ContentBackground = styled(Wrapper)`
+  flex: 1;
+  background-color: ${UIColors.defaultBackgroundColor};
+`;
+
+const genericToken = require('assets/images/tokens/genericToken.png');
+
+class SendSyntheticAsset extends React.Component<Props, State> {
+  state = {
+    loadingAssets: true,
+    availableAssets: [],
+  };
+  defaultSyntheticAsset: Asset;
+  syntheticServicePromise;
+
+  constructor(props: Props) {
+    super(props);
+    const { accountAssets, supportedAssets, availableStake } = props;
+    const assetsData = getAssetsAsList(accountAssets);
+    this.defaultSyntheticAsset = {
+      ...getAssetData(assetsData, supportedAssets, PLR),
+      balance: availableStake,
+      isSynthetic: true,
+    };
+  }
+
+  componentDidMount() {
+    this.props.initSyntheticsService();
+    this.getAvailableSyntheticsAssets();
+  }
+
+  componentWillUnmount() {
+    this.cancelSyntheticServiceRequest();
+  }
+
+  getAvailableSyntheticsAssets = () => {
```

Comment:
I think that part should be extracted into the action, as this is a business logic thing. Then you won't need a hack with cancelable promise